luarules/gadgets/unit_workertime_boost.lua

```diff
@@ -0,0 +1,67 @@
+function gadget:GetInfo()
+	return {
+		name = "Workertime Multiplier Boost",
+		desc = "Allows units with added UnitDefs to build gadget-defined types of units faster.",
+		author = "SethDGamre",
+		date = "April 2024",
+		license = "Public domain",
+		layer = 1,
+		enabled = true
+	}
+end
+
+-- synced only
+if not gadgetHandler:IsSyncedCode() then return false end
+
+-- is unba com on
+if true and Spring.GetModOptions().unbacom ~= "enabled" then return false end
+Spring.Echo("the script ran?!")
+Spring.Echo(Spring.GetModOptions().unbacom)
+local boosttriggers = {} -- stores what words parsed from the wtboostunittype "trigger" boost
+local mobileunits = {} -- stores the names of units that can move
+local boostedworkertimes = {}-- stores the values of builders who have defined workertimeboost definitions.
+local originalworkertimes = {} 
+-- workertimeboost = number -- in the unitdefs of the builder. This is the mulitplier by which workertime is boosted.
+-- wtboostunittype = defined in unitdef of builder, it's a table of strings such as "MOBILE" which defines what units boost buildpower for the builder.
+	
+--step 1, FOR make a table containing the  names of the units that can boost
+for id, def in pairs(UnitDefs) do
+	if def.buildSpeed then
+		if def.customParams.workertimeboost then
+			originalworkertimes[id] = def.buildSpeed
+			boostedworkertimes[id] = def.buildSpeed * def.customParams.workertimeboost--adds the key "id" unitname to the list. Boostable builders represented the boosted workertime.
+		end
+	end
+
+	if def.speed and def.speed ~= 0 then
+		mobileunits[id] = true
+	end
+
+	if def.customParams.wtboostunittype and string.find(def.customParams.wtboostunittype, "MOBILE") then
+		boosttriggers[id] = def.customParams.wtboostunittype
+	end
+end
```

Comment:
This type is set to the same thing for each com, is there a reason that isn't just set in this gadget instead?

Reply:
To leave potential for easy future expansion. When I do version 3 of unbalanced commanders, I might give boosts for specific types of structures or even specific units.  I won't know this until I get feedback from players about how well (or badly) unbacom plays in this iteration.